core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -4440,6 +4440,39 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     assert(mapStatuses.count(s => s != null && s.location.executorId == "hostB-exec") === 1)
   }
 
```

Comment:
Can we add a test specifically for the behavior we have introduced ? Namely, we send finalize shuffle even if sending msg to "first" merger blocks indefinitely (and so does not block a) other sends, b) merge finalization completes within the timeout).